lib/administrate/field/deferred.rb

```diff
@@ -30,9 +30,20 @@ def searchable?
       end
 
       def searchable_field
+        ActiveSupport::Deprecation.warn(
+          "searchable_field is deprecated, use searchable_fields instead",
+        )
         options.fetch(:searchable_field)
       end
 
+      def searchable_fields
+        if options.key?(:searchable_field)
+          [searchable_field]
```

Comment:
Am I right in thinking that the warning should also be shown here?

The method `searchable_field` is indeed deprecated, and it will be called by outdated code, so it's good that it shows the warning. On the other hand, a field defined before this change will define `:searchable_field` when it should be `:searchable_fields`. This method `searchable_fields` will be called by up-to-date code, making the translation, and no warning will be shown to users who used the deprecated option. I think it should be shown. What do you think?

Reply:
Logic here is a bit different. Existing user code that relies on older administrate versions will be using `:searchable_field` option for the `Filed::` associations. Method `searchable_field` is  unlikely to be used by users, this method is only called inside `Administrate::Search::Query`. With this patch `Administrate::Search::Query` will switch to using `searchable_fields`. So what we are trying to do with this code is provide support for old `:searchable_field` option inside updated `Administrate::Search::Query`. Hope this makes it a bit more clear.

Reply:
Yes, let me rephrase. Say that a user is doing the following:

```ruby
    parts: Field::HasMany.with_options(
      searchable: true,
      searchable_field: :name,
    ),
```

After merging this PR, they'll be using a deprecated option, `:searchable_field`, which should now be `:searchable_fields`:

```ruby
    parts: Field::HasMany.with_options(
      searchable: true,
      searchable_fields: [:name],
    ),
```

It's great that this MR provides this compatibility, but I think that eventually we should phase out this option as it won't be documented any more, and it will reduce maintenance costs going forward.

Therefore I think there should be a warning. Afterwards we can actually remove the option, probably after a couple of versions, so that the change is not too sudden on our users.

What do you think?

Reply:
I think this is already handled. If first example will be executed in the context of the new version the following will happen:

- Query related code will access `#searchable_fields`
- `#searchable_fields` will check that there is a `:searchable_field` key available in the options
- This will trigger `[searchable_field]` call
- `#searchable_field` will trigger deprecation warning

Reply:
Oh, of course. You are correct. Thank you for taking the time to explain this :+1: